tests/tries_test.go

```diff
@@ -0,0 +1,112 @@
+package tests
+
+import (
+	"fmt"
+	"math/big"
+	"math/rand"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/rawdb"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethdb/memorydb"
+	"github.com/ethereum/go-ethereum/trie"
+	"github.com/gballet/go-verkle"
+)
+
+func BenchmarkTriesRandom(b *testing.B) {
+	numAccounts := []int{1_000, 5_000, 10_000}
+
+	for _, numAccounts := range numAccounts {
+		rs := rand.New(rand.NewSource(42))
+		accounts := getRandomStateAccounts(rs, numAccounts)
+
+		b.Run(fmt.Sprintf("MPT/%d accounts", numAccounts), func(b *testing.B) {
+			trie, _ := trie.NewStateTrie(trie.TrieID(common.Hash{}), trie.NewDatabase(memorydb.New()))
+			b.ReportAllocs()
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				for k := 0; k < len(accounts); k++ {
+					trie.TryUpdateAccount(accounts[k].address[:], &accounts[k].stateAccount)
+				}
+				trie.Commit(true)
+			}
+		})
+		b.Run(fmt.Sprintf("VKT/%d accounts", numAccounts), func(b *testing.B) {
+			// Warmup VKT configuration
+			trie.NewVerkleTrie(verkle.New(), trie.NewDatabase(memorydb.New())).TryUpdate([]byte("00000000000000000000000000000012"), []byte("B"))
+
+			b.ReportAllocs()
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				trie := trie.NewVerkleTrie(verkle.New(), trie.NewDatabase(memorydb.New()))
+				for k := 0; k < len(accounts); k++ {
+					trie.TryUpdateAccount(accounts[k].address[:], &accounts[k].stateAccount)
+				}
+				trie.Commit(true)
+			}
+		})
+	}
+}
+
+type randomAccount struct {
+	address      common.Address
+	stateAccount types.StateAccount
+}
+
+func getRandomStateAccounts(rand *rand.Rand, count int) []randomAccount {
```

Comment:
Noting surprising here. Maybe at some point it might be interesting to add `CodeHash` values; but still there's a big gap with the current simple state accounts to look into.